docs/java/release-notes.mdx

```diff
@@ -27,21 +27,78 @@ We highly recommend regularly updating to the latest SDK version. It will help y
 - update client libraries giving access to latest SAP services on SAP Cloud Platform and S4/HANA
 - protect yourself from bugs and breaking changes in the future
 
+## 3.29.1 - September 24, 2020
+
+[Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*%20AND%20v:3.29.1)
+
+### Compatibility Notes
+
+- To leverage the new **OData v2 client**, you need to adjust existing code. Refer to [migration guide](./features/odata/use-typed-odata-v2-client-in-sap-cloud-sdk-for-java#switch-to-improved-odata-v2-type-safe-client) for detailed instructions.
+- We start deprecating APIs of the outdated **OData v2 client**. In this release we deprecated:
+  - Methods: `execute`, `toQuery`, `withErrorHandler`, `cachingMetadata`, `withoutCachingMetadata`, `ignoringVersionIdentifier` and `ignoreVersionIdentifier`.
+  - Error handler: `ODataVdmErrorResultHandler`.
+  - Exception classes `VersionIdentifierMissingException` and `VersionIdentifierExpiredException`.
+  - You'll find instructions how to adjust the code to use the new **OData v2 client** in the Javadoc of deprecated method or class.
+
+### New Functionality
+
+- The improved OData v2 type-safe client is now Generally Available. From now on we recommend it for OData API consumption.
+Refer to the [migration guide](./features/odata/use-typed-odata-v2-client-in-sap-cloud-sdk-for-java#switch-to-improved-odata-v2-type-safe-client) for more details on the new OData 2 client.
```

Comment:
Please, don't call it migration - it's a simple switch 😊

Reply:
We don't call it migration anymore, @sacnl. :)

Reply:
It reads "migration guide" in line 46 @artemkovalyov

Reply:
@Emdee89 , yeah, the push has failed because of the upstream merge. Take a look at it now.